tests/periph_cpuid/tests/01-run.py

```diff
@@ -15,7 +15,7 @@ def testfunc(child):
     child.expect_exact('This test is reading out the CPUID of the platforms CPU')
     child.expect(r'CPUID_LEN: (\d+)')
     cpuid_len = int(child.match.group(1))
-    child.expect(r'CPUID:( 0x[0-9a-fA-F]{2})+\s*$')
+    child.expect(r'CPUID:( 0x[0-9a-fA-F]{2})+\s*\r\n')
```

Comment:
This might break for terminal programs were the line break is not a `\r\n` but a `\n`. Why was not just `\n` used here?

Reply:
We need to document this somewhere, but a) pexpect states "wait for \r\n", and b) many tests depend on this.

https://pexpect.readthedocs.io/en/stable/overview.html#find-the-end-of-line-cr-lf-conventions

Reply:
Ok, I confirmed, that the test still works on current master.

Reply:
(with native, the most UNIXoid "TTY")

Reply:
Still, since the last pattern is `\s*`(an optional amount of whitespaces) just using `\n` should have worked as well.

Reply:
> We need to document this somewhere, but a) pexpect states "wait for \r\n", and b) many tests depend on this.
> 
> https://pexpect.readthedocs.io/en/stable/overview.html#find-the-end-of-line-cr-lf-conventions

Maybe a doc page "Writing tests" should be added.

Reply:
Or rather updating [Running and creating tests](https://doc.riot-os.org/running-and-creating-tests.html)